ampd/src/commands/register_public_key.rs

```diff
@@ -0,0 +1,51 @@
+use std::convert::TryFrom;
+use std::path::Path;
+
+use cosmrs::{cosmwasm::MsgExecuteContract, tx::Msg};
+use error_stack::{Result, ResultExt};
+use multisig::{
+    key::{KeyType, PublicKey},
+    msg::ExecuteMsg,
+};
+use report::ResultCompatExt;
+
+use crate::commands::{broadcast_tx, worker_pub_key};
+use crate::config::Config;
+use crate::{handlers, Error, PREFIX};
+
+pub async fn run(config: Config, state_path: &Path) -> Result<Option<String>, Error> {
+    let pub_key = worker_pub_key(state_path, config.tofnd_config.clone()).await?;
+
+    let multisig_address = config
+        .handlers
+        .iter()
+        .find_map(|config| {
+            if let handlers::config::Config::MultisigSigner { cosmwasm_contract } = config {
+                Some(cosmwasm_contract.clone())
+            } else {
+                None
+            }
+        })
+        .ok_or(Error::LoadConfig)
+        .attach_printable("no multisig contract found in config")?;
```

Comment:
does it make sense to make finding a specific config a function on the handler `Config`? Also, at this point it seems a bit weird to call these handler configs

Reply:
I also agree that it's weird to pull the handler config here.

Reply:
it's weird, can't find a good way atm. extracted to a helper function for now